### PR TITLE
Adapt intent matches not returned to core in confidence order.

### PR DESCRIPTION
**Pull request by chrisveilleux**
## Description
The AdaptService in core assumes that the highest confidence intent match will be the first result returned from Adapt. This is not always the case. It is possible that Adapt should be ordering the return value instead. But that would be a bigger change to Adapt.

NOTE: this replaces [PR2915](https://github.com/MycroftAI/mycroft-core/pull/2915)

## How to test
*  Remove the future time Padatious intent from the date/time skill
* Ask "what time will it be in ten minutes"; current time will be returned.
* Apply the change in this PR
* Make same request as before, future time will be returned.

## Contributor license agreement signed?
CLA [Y]

### CODE REVIEW

**Comment by JarbasAl** [comment]
isnt this handled by https://github.com/MycroftAI/adapt/pull/137 ?

merging that and bumping adapt seems like a better fix no?

File: mycroft/skills/intent_services/adapt_service.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def __init__(self, config):
         self.lock = Lock()
 
     def update_context(self, intent):
-        """Updates context with keyword from the intent.
+        """Updates context wit4h keyword from the intent.
```
Comment (JarbasAl) [inline mycroft/skills/intent_services/adapt_service.py:183]: typo

Comment (chrisveilleux) [reply mycroft/skills/intent_services/adapt_service.py:183]: Thank you sir.

**Comment by chrisveilleux** [comment]
> isnt this handled by [MycroftAI/adapt#137](https://github.com/MycroftAI/adapt/pull/137) ?
> 
> merging that and bumping adapt seems like a better fix no?

I am not entirely sure the code in that PR fixes this issue.

**Comment by clusterfudge** [comment]
> > isnt this handled by [MycroftAI/adapt#137](https://github.com/MycroftAI/adapt/pull/137) ?
> > merging that and bumping adapt seems like a better fix no?
> 
> I am not entirely sure the code in that PR fixes this issue.

@chrisveilleux can you clarify what edge cases you think [MycroftAI/adapt#137](https://github.com/MycroftAI/adapt/pull/137) does not address? happy to add tests to confirm/fix.

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2947)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/core/PR-2947/logs/skills.log), [audio.log](https://reports.mycroft.ai/core/PR-2947/logs/audio.log), [voice.log](https://reports.mycroft.ai/core/PR-2947/logs/voice.log), [bus.log](https://reports.mycroft.ai/core/PR-2947/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/core/PR-2947/logs/enclosure.log)

**Comment by chrisveilleux** [comment]
> > > isnt this handled by [MycroftAI/adapt#137](https://github.com/MycroftAI/adapt/pull/137) ?
> > > merging that and bumping adapt seems like a better fix no?
> > 
> > 
> > I am not entirely sure the code in that PR fixes this issue.
> 
> @chrisveilleux can you clarify what edge cases you think [MycroftAI/adapt#137](https://github.com/MycroftAI/adapt/pull/137) does not address? happy to add tests to confirm/fix.

I will try to give an example in the Adapt PR.

**Comment by chrisveilleux** [comment]
Closing this PR.  I believe the issue was fixed with [Adapt PR 128](https://github.com/MycroftAI/adapt/pull/128).  The parse confidence passed to `Intent.validate_with_tags()` was overridden in the method, causing an incorrect value to be used in the final confidence calculation.

I tested the scenario that caused me to generate this PR and it is no longer an issue.

A great example of why you should never mess with the value of a parameter in a method or function.
